### PR TITLE
Notebook on history usage and comparison of multiple results.

### PR DESCRIPTION
**Pull request by PaulJonasJost**
Due to a recent question, I realised that we don't have any example on history usage (not that I found it) beyond the storage and loading. Added a small notebook to highlight some usacases for a history.

### CODE REVIEW

**Comment by dweindl** [comment]
Wouldn't some non-ODE model (Rosenbrock, ...) be sufficient here?
1) faster
2) fewer dependencies
3) doesn't confuse non-amici/petab users

File: doc/example/history_usage.ipynb [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+   "source": [
+    "A final note on the `Hdf5History` together with the `MultiProcessEngine`.\n",
+    "\n",
+    "If you use both together, the way the history works is the following:\n",
+    "- You specified a history file `fn_history` to store the result in\n",
+    "- A folder with the name `fn_history` will be created at the start of optimization\n",
+    "- Each optimization gets its own separate Hdf5 file within the folder named `fn_history_{id}`\n",
+    "- **After** optimization, the `fn_history` file will be created such that the folder is in the same directory and each result will be linked within this file, such that you can easily work with the `fn_history` file, that pools together all other histories\n",
+    "\n",
+    "This also means, that if your optimization is interrupted, there is **no fn_history** file for pooling. In this case the `optimization_result_from_history` does not work nicely for all of them together. In this case you can load each result separately through the `optimization_result_from_history` and append the `optimize_result`'s together."
```
[the remark this answers is not in the document]
Comment (Doresic) [reply doc/example/history_usage.ipynb:417]: In the last sentence, should it be `optimization_result_from_history `? Or `read_results_from_file`? 
I think the loading each individual and combining them is done using the `read_results_from_file` which calls `read_result_from_file` for each individual available history, no?

**Review by Doresic — APPROVED** [review]
Nice notebook, good descriptions, fast. 
Would nice to have some further details of how to load the Result objective if the central history file was not created. 
A notebook describing this was definitely missing, and will be a nice addition.
Thanks!

File: doc/example/history_usage.ipynb [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+   "metadata": {},
+   "source": [
+    "## Reloading from History"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Especially when running large models on clusters, the optimization sometimes my stop due to unfortunate reasons (e.g. timeouts). In these cases, the history serves yet another purpose: retrieving finished and unfinished optimizations. Sometimes out of 100 starts, 80 might have already been terminated, in this case, investigating those 80 might already yield good results. In other cases, we might want to continue optimization from where we left of."
```
[the remark this answers is not in the document]
Comment (PaulJonasJost) [reply doc/example/history_usage.ipynb:336]: it is true for a SingleCoreEngine and it is in general true for the optimization. as I mention below you can use the `optimization_result_from_history` for a singular start (even if the central file) is non existent. you use it on the files within the folder. the drawback is here that each file gives you a separate `Result` object, which you would need to merge.

File: doc/example/history_usage.ipynb [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+   "outputs": [],
+   "source": [
+    "result_from_history.optimize_result[0].history.get_fval_trace() == result_fides.optimize_result[0].history.get_fval_trace()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Important to note here is, that we did **not use the result** saved in hdf5 to fill in the loaded result, but solely the history. This means, that if you use an `Hdf5History` during optimization and the optimization is interrupted, you can still load the optimization up to that point. From there you can for example restart the non finished (lets say e.g. the last 3) starts and finish optimizing them:"
```
Comment (PaulJonasJost) [inline doc/example/history_usage.ipynb:385]:
```suggestion
    "Important to note here is, that we did **not use the result** saved in hdf5 to fill in the loaded result, but solely the history. This means, that if you use an `Hdf5History` during optimization and the optimization is interrupted, you can still load the optimization up to that point. From there you can, for example, restart the non finished (lets say e.g. the last 3) starts and finish optimizing them:"
```

File: doc/example/history_usage.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (PaulJonasJost) [reply doc/example/history_usage.ipynb:417]: `read_results_from_file` requires the HistoryOptions, and it assumes that all histories are in the same file. The `optimization_result_from_history` just takes the hdf5-file, and then within it looks for all histories it has. I feel that if we have a hdf5 file, this is the more appropriate way of loading it. I did not add an example for each file separately as it is functionally exactly the same as the first one, just looped through the files and I was not quite sure how to artificially create the folder with each file temporarily. If you think this has a lot of merit, will try to add it here.

**Review by Doresic — APPROVED** [review]
Much clearer! Thanks!

**Pull request merged**